tokio/src/io/read_buf.rs

```diff
@@ -55,6 +59,47 @@ impl<'a> ReadBuf<'a> {
         }
     }
 
+    cfg_io_util! {
+         fn from_buf(buf: &'a mut impl bytes::BufMut) -> ReadBuf<'a> {
+            let buf = buf.chunk_mut();
+            ReadBuf {
+                // SAFETY: ReadBuf guarantees that no uninitialized bytes are written to `buf`
```

Comment:
Should this method be changed to take an [`UninitSlice`][1] as argument instead?

[1]: https://docs.rs/bytes/1/bytes/buf/struct.UninitSlice.html